File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP5_projet_1_Sofiane/Projet_1_MEDDOURI_VANSCOOR_premier/Voiture.java
import java.util.Date;

public class Voiture{
    
    // attributs

    private String numeroIdentification;
    protected Modele modele;
    protected Moteur moteur;
    private Date dateFabrication;
    private Couleur couleur;
    private Options[] options;
   

    // constructeur par défaut

    public Voiture(){
        this.numeroIdentification = "";
        this.modele = null;
        this.moteur = null;
        this.couleur = Couleur.BLEU;
        this.options = null;
        
    }

    // constructeur par initialisation

	public Voiture(String numeroIdentification, Modele modele, Moteur moteur, Date dateFabrication, Couleur couleur,Options options[]){
		this.numeroIdentification = numeroIdentification;
		this.modele = modele;
		this.moteur = moteur;
		this.dateFabrication = dateFabrication;
        this.couleur = couleur;
        this.options = options;
    }
    
    // constructeur par copie

    public Voiture(Voiture v){
        this.numeroIdentification = v.numeroIdentification;
        this.modele = v.modele;
        this.moteur = v.moteur;
        this.dateFabrication = v.dateFabrication;
        this.couleur = v.couleur;
        this.options = v.options;

    }

    // getters

    public String getNumeroIdentification(){
        return this.numeroIdentification;
    }

    public Moteur getMoteur(){
        return this.moteur;
    }

    public Modele getModele(){
        return this.modele;
    }

    public Date getDateFabrication(){
        return this.dateFabrication;
    }

    public Couleur getCouleur(){
        return this.couleur;
    }

    public Options[] getOptions(){
        return this.options;
    }

    public String getStringOptions(){
        String o = "";
        for(int i = 0; i < this.options.length;i++){
            o += this.options[i].getNomOptions() + ", ";
        }
        return o;
    }

    public String toString(){
        return "numeroIdentification : " + getNumeroIdentification() + "\nmoteur :" + getMoteur() + " modele : " + getModele() +  "\ndateFabrication : " + getDateFabrication() + "\ncouleur : " + getCouleur() + "\noptions : " + getStringOptions();
    }





}

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2_sofiane/Voiture.php
<?php

class Voiture implements JsonSerializable{
    private $numeroIdentification;
    private $modele;
    private $moteur;
    private $dateFabrication;
    private $couleur;
    private $options;

    // constructeur

    public function __construct($numeroIdentification,$modele,$moteur,$dateFabrication,$couleur,$options){
    	$this->numeroIdentification = $numeroIdentification;
        $this->modele = $modele;
        $this->moteur = $moteur;
        $this->dateFabrication = $dateFabrication;
        $this->couleur = $couleur;
        $this->options = $options;
    }

    

    // getters

    public function getNumeroIdentification(){
        return $this->numeroIdentification;
    }

    public function getMoteur() {
        return $this->moteur;
    }

    public function getModele(){
        return $this->modele;
    }

    public function getDateFabrication() {
        return $this->dateFabrication;
    }

    public function getCouleur() {
        return $this->couleur;
    }

    public function getOptions() {
        return $this->options;
    }

    // methode toString

    public function __toString(){
        return "numeroIdentification : " . $this->numeroIdentification . " modele : " . $this->modele . " moteur : " . $this->moteur . " couleur : " . $this->couleur . " options : " . $this->options;
    }

    public function jsonSerialize() : array{
        return ['numeroIdentification' => $this->numeroIdentification, 'modele'=> $this->modele, 'moteur' => $this->moteur, 'dateFabrication' => $this->dateFabrication, 'couleur' => $this->couleur, 'options'=> $this->options];
    }

    public static function FromJson($json){
        $table = json_decode($json);
        return new self($table->numeroIdentification,$table->modele,$table->moteur,$table->dateFabrication,$table->couleur,$table->options);
    }
}
?>